can/interfaces/kvaser/canlib.py

```diff
@@ -468,22 +493,43 @@ def __init__(self, channel, can_filters=None, **kwargs):
             ctypes.byref(ctypes.c_long(TIMESTAMP_RESOLUTION)),
             4,
         )
-
-        if fd:
-            if "tseg1" not in kwargs and bitrate in BITRATE_FD:
-                # Use predefined bitrate for arbitration
-                bitrate = BITRATE_FD[bitrate]
-            if data_bitrate in BITRATE_FD:
-                # Use predefined bitrate for data
-                data_bitrate = BITRATE_FD[data_bitrate]
-            elif not data_bitrate:
-                # Use same bitrate for arbitration and data phase
-                data_bitrate = bitrate
-            canSetBusParamsFd(self._read_handle, data_bitrate, tseg1, tseg2, sjw)
+        if isinstance(timing, BitTimingFd):
+            timing = check_or_adjust_timing_clock(timing, [80_000_000])
+            canSetBusParams(
+                self._read_handle,
+                timing.nom_bitrate,
+                timing.nom_tseg1,
+                timing.nom_tseg2,
+                timing.nom_sjw,
+                1,
+                0,
+            )
+            canSetBusParamsFd(
+                self._read_handle,
+                timing.data_bitrate,
+                timing.data_tseg1,
+                timing.data_tseg2,
+                timing.data_sjw,
+            )
+        elif isinstance(timing, BitTiming):
+            timing = check_or_adjust_timing_clock(timing, [16_000_000])
+            canSetBusParamsC200(self._read_handle, timing.btr0, timing.btr1)
         else:
-            if "tseg1" not in kwargs and bitrate in BITRATE_OBJS:
-                bitrate = BITRATE_OBJS[bitrate]
-        canSetBusParams(self._read_handle, bitrate, tseg1, tseg2, sjw, no_samp, 0)
+            if fd:
+                if "tseg1" not in kwargs and bitrate in BITRATE_FD:
+                    # Use predefined bitrate for arbitration
+                    bitrate = BITRATE_FD[bitrate]
+                if data_bitrate in BITRATE_FD:
+                    # Use predefined bitrate for data
+                    data_bitrate = BITRATE_FD[data_bitrate]
+                elif not data_bitrate:
+                    # Use same bitrate for arbitration and data phase
+                    data_bitrate = bitrate
+                canSetBusParamsFd(self._read_handle, data_bitrate, tseg1, tseg2, sjw)
+            else:
+                if "tseg1" not in kwargs and bitrate in BITRATE_OBJS:
```

Comment:
This is unrelated to your PR, but do they always use the same `tseg1` / `tseg2` for both nominal and data phases?

Reply:
My guess is, if you use the predefined constants (e.g. `canFD_BITRATE_500K_80P = -1000`), then these parameters are irrelevant and the sample point will always be at 80%